### PR TITLE
Add assembly versioning targets to the build tools package.

### PR DESCRIPTION
**Pull request by weshaggard**
These target files will generate the necessary _AssemblyInfo file
with the various assembly level attributes, like assembly and file
version, as well as name and copyright, etc.

### CODE REVIEW

**Comment by weshaggard** [comment]
cc @chcosta @jhendrixMSFT

File: src/nuget/Microsoft.DotNet.BuildTools.nuspec [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <package>
   <metadata minClientVersion="2.8.1">
     <id>Microsoft.DotNet.BuildTools</id>
-    <version>1.0.25-prerelease</version>
+    <version>1.0.26-prerelease</version>
```
Comment (AlexGhiondea) [inline src/nuget/Microsoft.DotNet.BuildTools.nuspec]: With the new versioning scheme should this be prerelease-1?

Comment (jhendrixMSFT) [inline src/nuget/Microsoft.DotNet.BuildTools.nuspec]: The build process will automatically append the "dash daily build number" to the package version so no need to include it directly.

Comment (weshaggard) [inline src/nuget/Microsoft.DotNet.BuildTools.nuspec]: Forgot about this new build number scheme I will revert this change.

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Pull request merged**